### PR TITLE
Update `contact_matrix` to only accept `survey()` objects

### LINKED ISSUE #93: Accept only `survey` objects in `contact_matrix()`
Currently, the `survey` argument of `contact_matrix()` is passed to `get_survey()`, which means it can be a DOI or an URL.

This:

- does not correspond to the documentation, which states `survey` must be a `survey` object
- provides different ways to do the same thing (`download_survey(doi) |> load_survey() |> contact_matrix()` vs `contact_matrix(doi)`), which is generally a code smell since it makes it more difficult to maintain the code and for users to get familiar with the packages

I recommend to update `contact_matrix()` to match the documented behaviour. Once this is done, the diversion through `get_survey()` in `contact_matrix()` is no longer necessary and `get_survey()` should probably ultimately be deprecated.

### PR DESCRIPTION
**Pull request by sbfnk**
Closes #93

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # socialmixr 0.3.1.9000
 
 * An error in `list_surveys()` was fixed which stopped this working
+* `contact_matrix` was updated to only accept `survey()` objects, not DOIs.
```
Comment (Bisaloo) [inline NEWS.md:4]:
````suggestion
* `contact_matrix()` was updated to only accept `survey` objects, not DOIs and matches the documentation. It is still possible to get a contact matrix from a DOI but it is necessary to go through the `get_survey()` function.

  ```r
  # No longer works!
  contact_matrix("10.5281/zenodo.1095664")
  
  # Recommended workflow
  get_survey("10.5281/zenodo.1095664") |>
    contact_matrix()
  ```
````

**Review by Bisaloo — COMMENTED** [review]
Should this go through a formal deprecation process?

**Comment by sbfnk** [comment]
> Should this go through a formal deprecation process?

Yes, that is a very good point.

File: R/contact_matrix.r [hunk 2]
```diff
@@ -56,8 +57,24 @@ contact_matrix <- function(survey, countries = NULL, survey.pop, age.limits, fil
...
+  if (all(is_doi(survey))) {
+    deprecate_warn(
+      "0.3.2", paste0("contact_matrix(survey = '", error_string, "')"),
+      details = "Passing a DOI will be removed in version 0.4.0."
+    )
+    survey <- get_survey(survey)
+  } else if (!inherits(survey, "survey")) {
+    stop(
+      "'survey' must be a survey object (created using)",
+      "or `get_survey()`)."
```
Comment (Bisaloo) [inline R/contact_matrix.r:72]: This message was accidentally modified in the addition of the deprecation process.

```suggestion
      "'survey' must be a survey object (created using `survey()`",
      " or `get_survey()`)."
```

Comment (sbfnk) [reply R/contact_matrix.r]: addressed by using a common error string in ed8d823

**Pull request merged**